_core/smart/tutorial-counter.md

```diff
@@ -0,0 +1,177 @@
+? Project name: (clarity-counter)
+```
+
+Finally, the project dependencies are installed and your project is ready for development. Because you already completed the Hello World tutorial, the project structure is familiar to you. The main difference is that we have additional tests for a new counter smart contract.
+
+## Step 2: Running tests
+
+Smart contracts are often developed in a test-driven approach to ensure code quality but also to speed up the development cycle by removing the need to push every change to the Blockchain before executing it. We will do the same in this project. Now, let's run the tests and review the results:
+
+Still in the project root directory, run the following command:
+
+```bash
+npm test
+```
+
+You should see the following response:
+
+```bash
+counter contract test suite
+    ✓ should have a valid syntax
+    deploying an instance of the contract
+    1) should start at zero
+    2) should increment
+    3) should decrement
+
+
+1 passing (734ms)
+3 failing
+```
+
+It looks like we see some failed test! That is on purpose - we will implement the new smart contract in the next steps! After every increment of the contract, we will rerun the tests to ensure we're on the right track.
+
+## Step 3: Developing a smart contract
+
+Let's get familiar with the tests to understand what the new smart contract should look like
+
+1. Take a quick look at the test file associated with the counter smart contract:
+
+    ```shell
+    cat test/counter.ts
+    ```
+
+    You should be familiar with the test set up from the Hello World tutorial. Notice how the instance of the smart contract is created on line 8:
+
+    ```js
+    counterClient = new Client("SP3GWX3NE58KXHESRYE4DYQ1S31PQJTCRXB3PE9SB.counter", "counter", provider);
+    ```
+
+    That tells us that the new smart contract is named `counter` and that it should be found in the following file: `contracts/counter.clar`. Note that the `contracts` folder is assumed as the base folder and that every Clarity file has the suffix `.clar`.
+
+    The file was already created during the project setup.
+
+2. With the editor of your choice, open the file and add the following lines of code:
+
+    ```cl
+    (define-data-var counter int 0)
+
+    (define-public (get-counter)
+      (ok (var-get counter))
+    )
+    ```
+
+    The first line initializes a new integer variable `counter` with the value set to `0` using the [`define-data-var`](https://docs.blockstack.org/core/smart/clarityref#define-data-var) statement. It is important to note that all definition statements in Clarity need to be at the top of the file.
+
+    To provide access to the `counter` variable from outside of the current smart contract, we need to declare a public get function. The last lines of the code add a public `get-counter` function. The [`var-get`](https://docs.blockstack.org/core/smart/clarityref#var-get) statement looks for a variable in the contract's data map and returns it.
```

Comment:
`declare a public get function` --> `declare a public function to get it`.

Reply:
Also, `data map` --> `data space`.  We need to be careful with the word `map`, since Clarity has a notion of data variables and data maps (both are part of the data space of a contract).